### PR TITLE
[HUD] add percentage for linux rollover

### PR DESCRIPTION
**Pull request by PaliC**
Adds a graph showing the percentage of jobs on the linux foundation

### CODE REVIEW

**Comment by clee2000** [comment]
Any chance I could get you to use ClickHouse instead of Rockset?

File: torchci/pages/metrics.tsx [hunk 1]
```diff
@@ -1133,6 +1133,18 @@ export default function Page() {
...
+        <Grid item xs={12} height={ROW_HEIGHT}>
+          <TimeSeriesPanel
+            title={"Percentage of jobs rolled over to Linux Foundation"}
+            queryName={"lf_rollover_percentage"}
+            queryCollection={"metrics"}
+            queryParams={[]}
+            granularity={"day"}
+            timeFieldName={"bucket"}
+            yAxisFieldName={"percentage_lf"}
+            yAxisRenderer={(value) => value + "%"}
```
Comment (ZainRizvi) [inline torchci/pages/metrics.tsx:1145]: the rendering didn't work.  You might need to convert `value` to a string first

image

File: torchci/rockset/metrics/__sql/lf_rollover_percentage.sql [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                    WHEN conclusion = 'failure' THEN duration_min
+                    ELSE 0
+                END
+            ) / COUNT(*) as failure_avg_duration,
+            SUM(
+                CASE
+                    WHEN conclusion = 'cancelled' THEN duration_min
+                    ELSE 0
+                END
+            ) / COUNT(*) as cancelled_avg_duration,
```
Comment (ZainRizvi) [inline torchci/rockset/metrics/__sql/lf_rollover_percentage.sql:136]: you don't seem to need these, so might as well delete them

File: torchci/rockset/metrics/__sql/lf_rollover_percentage.sql [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            label
+    ),
+    comparison_stats AS (
+        SELECT
+            lf.bucket,
+            SUM(lf.group_size + m.group_size) as total_jobs,
+            SUM(m.group_size) as meta,
+            SUM(lf.group_size) as lf,
+            CAST(SUM(m.group_size) as FLOAT) / SUM(lf.group_size + m.group_size) * 100 as percentage_meta,
+            CAST(SUM(lf.group_size) as FLOAT) / SUM(lf.group_size + m.group_size) * 100 as percentage_lf
```
Comment (ZainRizvi) [inline torchci/rockset/metrics/__sql/lf_rollover_percentage.sql:152]: to be able to render both `percentage_meta` and `percentage_lf` on the same chart, you'll need a single column to contain all the percentages, and another column to say if it's a meta or lf result (let's call cat `fleet`).  Then in the metrics page you'll set `groupByFieldName` to `fleet`.

That'll result in a nice side-by-side comparison, where you can see one fleet going up in usage as the other goes down

**Review by ZainRizvi — CHANGES_REQUESTED** [review]

File: torchci/rockset/metrics/__sql/lf_rollover_percentage.sql (hunk 2 above)
Comment (ZainRizvi) [inline torchci/rockset/metrics/__sql/lf_rollover_percentage.sql:136]: Any reason to keep these?

File: torchci/rockset/metrics/__sql/lf_rollover_percentage.sql (hunk 3 above)
Comment (PaliC) [reply torchci/rockset/metrics/__sql/lf_rollover_percentage.sql:152]: I think by itself percentage_lf is sufficient especially at the moment. However, we can chagne this later

File: torchci/pages/metrics.tsx (hunk 1 above)
Comment (PaliC) [reply torchci/pages/metrics.tsx:1145]: This is interesting, as it works on the side, but will fix

File: torchci/rockset/metrics/__sql/lf_rollover_percentage.sql (hunk 2 above)
Comment (PaliC) [reply torchci/rockset/metrics/__sql/lf_rollover_percentage.sql:136]: nope

File: torchci/pages/metrics.tsx (hunk 1 above)
Comment (ZainRizvi) [reply torchci/pages/metrics.tsx]: Really? The vercel preview didn't have it working on the side either. Sounds like worked locally for you though

Comment (ZainRizvi) [reply torchci/pages/metrics.tsx]: Vercel still doesn't show render the value :/ 
See https://torchci-git-fork-palic-palic-lfpercent-fbopensource.vercel.app/metrics

File: torchci/pages/metrics.tsx [hunk 4]
```diff
@@ -1133,16 +1133,17 @@ export default function Page() {
...
           />
         </Grid>
-        <Grid item xs={12} height={ROW_HEIGHT}>
+        <Grid item xs={6} height={ROW_HEIGHT}>
           <TimeSeriesPanel
             title={"Percentage of jobs rolled over to Linux Foundation"}
             queryName={"lf_rollover_percentage"}
             queryCollection={"metrics"}
             queryParams={[]}
             granularity={"day"}
```
Comment (ZainRizvi) [inline torchci/pages/metrics.tsx:1142]: `hour` granularity would be really nice here

**Review by ZainRizvi — APPROVED** [review]

**Pull request merged**